frontend/pages/_document.tsx

```diff
@@ -17,6 +17,18 @@ class MyDocument extends Document {
                   src={`${polyfillDomain}/v3/polyfill.min.js?features=default,Intl.RelativeTimeFormat,Intl.RelativeTimeFormat.~locale.en,Object.fromEntries,ResizeObserver`}
                   strategy="beforeInteractive"
                />
+               <Script
+                  strategy="beforeInteractive"
+                  src={`https://www.googletagmanager.com/gtag/js?id=${GTAG_ID}`}
+               ></Script>
```

Comment:
its one of the advantages of having this in `_document` is we can take advantage of preloading it. I tested it a bit with lighthouse and results seemed to be the same or slightly better with it being `beforeInteractive`.

Reply:
The Next.js docs say
![image](https://github.com/iFixit/react-commerce/assets/52104630/662cc297-d9f8-4569-9a72-95d9f33887a1)

I feel strongly that we should follow the recommended practice here to reduce TTI. I don't see how blocking hydration to load a tag manager script could make TTI better.

Reply:
The issue I ran to is that
```
<Script id="gtag-ga4" strategy="beforeInteractive">
                  {`
                  window.dataLayer = window.dataLayer || [];
                  function gtag(){dataLayer.push(arguments);}
                  gtag('js', new Date());
                  gtag('config', '${GTAG_ID}', ${GA_DEBUG} ? { debug_mode: true } : {});
               `}
               </Script>
```

Should be loaded immediately so it can exist for anyone that wants to call gtag. the best way i found to guarantee that is by putting it in `_documents` on `beforeInteractive`.

 > I don't see how blocking hydration to load a tag manager script could make TTI better.
 considering the script gets cached on the frontend, i imagine the backend has to be doing the same in this case

Reply:
that said, if you really want we can switch the `src={`https://www.googletagmanager.com/gtag/js?id=${GTAG_ID}`}` to `afterInteractive` again. I just don't think we should do it with the other analytics script.

Reply:
> that said, if you really want we can switch the `src={`[https://www.googletagmanager.com/gtag/js?id=${GTAG_ID}`}`](https://www.googletagmanager.com/gtag/js?id=$%7BGTAG_ID%7D%60%7D%60) to `afterInteractive` again. I just don't think we should do it with the other analytics script.

That's all I'm suggesting, is to make the googletagmanager.com script `afterInteractive`.

The other one `id="gtag-ga4"` was `beforeInteractive` previously, and I agree that it should remain that way. It runs a finite amount of JS and it initializes a global that other JS depends on, whereas the googletagmanager.com has an undefined amount of JS to run.

Regarding caching, I'm not exactly sure how changing `afterInteractive` to `beforeInteractive` would effect caching.